src/lib/markets/graph.ts

```diff
@@ -709,6 +709,143 @@ export async function fetchSubPositions({
+  address: Address
+  fromTs?: bigint
+  toTs?: bigint
+}) {
+  const maxTimeRange = BigInt(7 * 24 * 60 * 60)
+  const now = BigInt(Math.floor(Date.now() / 1000))
+  if (!fromTs) fromTs = now - maxTimeRange
+  if (!toTs) toTs = now
+
+  if (toTs - fromTs > maxTimeRange) {
+    throw new Error('The time range exceeds the maximum allowed range of 7 days.')
+  }
+  // TODO(arjun): update this query to use trades entity in subgraph when available
+  const tradeHistoryQuery = gql(`
+  query fetchTradeHistory($account: Bytes!, $fromTs: BigInt, $toTs: BigInt) {
+   accountPositionProcesseds(
+     where: { account: $account, blockTimestamp_gte: $fromTs, blockTimestamp_lte: $toTs },
+     orderBy: toOracleVersion,
+     orderDirection: desc
+   ) {
+     accumulationResult_collateralAmount, accumulationResult_keeper, accumulationResult_positionFee, priceImpactFee
+       accumulatedPnl, accumulatedFunding, accumulatedInterest, accumulatedMakerPositionFee, accumulatedValue
+       side, size, fromOracleVersion, toOracleVersion, toVersionPrice, toVersionValid, collateral, blockNumber,
+       market
+     update {
+       version, collateral, newMaker, newLong, newShort, valid, transactionHash, price, priceImpactFee,
+       localPositionId, globalPositionId, market, account, blockNumber, blockTimestamp, protect, interfaceFee, orderFee, side, delta
+     }
+   }
+  }
+ `)
+
+  const { accountPositionProcesseds } = await graphClient.request(tradeHistoryQuery, {
+    account: address,
+    fromTs: fromTs.toString(),
+    toTs: toTs.toString(),
+  })
+
+  const marketAccountPositionProcesseds = accountPositionProcesseds.reduce(
+    (acc, entity) => {
+      const market = getAddress(entity.market)
+      if (!acc[market]) acc[market] = []
+      acc[market].push(entity)
+      return acc
+    },
+    {} as Record<Address, typeof accountPositionProcesseds>,
+  )
+
+  type Update = (typeof accountPositionProcesseds)[number]['update'] & {
+    accountPositionProcesseds: typeof accountPositionProcesseds
+  }
+
+  const trades = Object.entries(marketAccountPositionProcesseds).reduce(
+    (acc, [market, entities]) => {
+      const updates: Update[] = []
+      entities.forEach((entity) => {
+        if (
+          entity.update &&
+          entity.update.side !== 'none' &&
+          entity.update.delta !== '0' &&
+          entity.update.collateral !== '0'
+        ) {
+          updates.push({ ...entity.update, accountPositionProcesseds: [entity] })
+        } else {
+          const update =
+            updates[updates.length - 1] ??
+            entities.find(
+              (e) => e.update && e.update.side !== 'none' && e.update.delta !== '0' && e.update.collateral !== '0',
+            )
+          if (update) {
+            update.accountPositionProcesseds.push(entity)
+          }
+        }
+      })
+
+      acc[getAddress(market)] = updates.map((update, i, self) => {
+        const accumulations = update.accountPositionProcesseds.filter(
```

Comment:
We shouldn't need this filter since the accountPositionProcesseds should already be correctly processed by the above step